src/main/java/com/thoughtworks/gauge/Table.java

```diff
@@ -209,14 +203,44 @@ private int getMaxStringSize(List<String> candidates) {
         return new Comparator<String>() {
 
             @Override
-            public int compare(
-                    String o1,
-                    String o2) {
+            public int compare(String o1, String o2) {
                 if (o1.length() < o2.length()) {
                     return -1;
                 }
                 return 1;
             }
         };
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((headers == null) ? 0 : headers.hashCode());
+        result = prime * result + ((rows == null) ? 0 : rows.hashCode());
+        result = prime * result + ((tableRows == null) ? 0 : tableRows.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        Table other = (Table) obj;
+        // Check for headers
+        if (!Arrays.equals(headers.toArray(), other.getColumnNames().toArray())) {
+            return false;
+        }
+        if (!tableRows.equals(other.getTableRows())) {
```

Comment:
Or simplify the whole thing to `return headers.equals(other.getColumnNames()) && tableRows.equals(other.getTableRows());`